landscape/client/manager/shutdownmanager.py

```diff
@@ -58,9 +58,31 @@ def _respond_success(self, data, operation_id):
             lambda _: self.registry.broker.stop_exchanger())
         return deferred
 
-    def _respond_failure(self, failure, operation_id):
+    def _respond_failure(self, failure, operation_id, reboot):
         logging.info("Shutdown request failed.")
-        return self._respond(FAILED, failure.value.data, operation_id)
+        failure_report = '\n'.join([
+            failure.value.data,
+            "",
+            "Attempting to force %(operation)s. Please note that if this "
+            "succeeds, Landscape will have no way of knowing and will still "
+            "mark this activity as having failed. It is recommended you check "
+            "the state of the machine manually to determine whether "
+            "%(operation)s succeeded." %
+            {"operation": "reboot" if reboot else "shutdown"}
+        ])
+        deferred = self._respond(FAILED, failure_report, operation_id)
+        # Add another callback spawning the poweroff or reboot command (which
+        # seem more reliable in aberrant situations like a post-trusty release
+        # upgrade where upstart has been replaced with systemd). If this
+        # succeeds, we won't have any opportunity to report it and if it fails
+        # we'll already have responded indicating we're attempting to force
+        # the operation so either way there's no sense capturing output
```

Comment:
I disagree: a log entry on failure would be better than dropping the input. But yeah, on success, it goes down too fast to handle any result.

Reply:
Yeah, that's sort of what I'm alluding to in that comment; if we delay reporting the status until we actually try reboot / poweroff then in the case of success we won't have any opportunity to report it, in which case we'd successfully report failure and fail to report success :) Hence, I think the current situation is preferable: always report, but warn in the case of failure that it might actually be success.